### PR TITLE
fix: add number of posts

### PR DESCRIPTION
**Pull request by sk-tenba**
https://app.asana.com/0/home/1203849283056222/1204124790686694

### CODE REVIEW

**Comment by sk-tenba** [comment]
image

File: containers/PageProjectDetails/containers/ProjectDetails/components/TabControl/index.module.scss [hunk 1]
```diff
@@ -58,3 +58,9 @@
 .tab + .tab {
   border-left: 1px solid rgba(0, 0, 0, 0.1);
 }
+
+.numOfAnnouncements {
+  position: absolute;
+  left: 38px;
+  top: 15px;
+}
```
Comment (sk-kitsune) [inline containers/PageProjectDetails/containers/ProjectDetails/components/TabControl/index.module.scss:66]: resize the browser --> ui breaks

----

🔴 **Blocking issue**: this PR cannot be merged until this issue is resolved.

Comment (sk-kitsune) [reply containers/PageProjectDetails/containers/ProjectDetails/components/TabControl/index.module.scss:66]: why using these magic numbers?

File: containers/PageProjectDetails/containers/ProjectDetails/components/TabControl/index.tsx [hunk 2]
```diff
@@ -3,53 +3,118 @@ import * as React from "react";
...
+import Typography from "@/modules/teiki-ui/components/Typography";
 
 type Props = {
   className?: string;
-  tabs: Tab[];
+  style?: React.CSSProperties;
   value: number;
-  onChange: (newValue: number) => void;
+  onChange: (newValue: number, hash: string) => void;
+  numOfPosts: number;
```
Comment (sk-kitsune) [inline containers/PageProjectDetails/containers/ProjectDetails/components/TabControl/index.tsx:13]: The component name is `TabControl`, not `TabControlForPosts`

Do not name your prop `numOfPosts`.

----

🔴 **Blocking issue**: this PR cannot be merged until this issue is resolved.

File: containers/PageProjectDetails/containers/ProjectDetails/components/TabControl/index.tsx [hunk 3]
```diff
@@ -3,53 +3,118 @@ import * as React from "react";
 
 import styles from "./index.module.scss";
 
-type Tab = {
-  title: React.ReactNode;
-  hash: string;
-  disabled?: boolean;
-};
```
Comment (sk-kitsune) [inline containers/PageProjectDetails/containers/ProjectDetails/components/TabControl/index.tsx:10]: The original code looks clearer.

To show the number, simply add another field `count: number` or `itemCount: number` in the definition of `Tab`.

File: containers/PageProjectDetails/containers/ProjectDetails/components/TabControl/index.tsx [hunk 4]
```diff
@@ -3,53 +3,118 @@ import * as React from "react";
...
 };
 
 export default function TabControl({
   className,
-  tabs,
+  style,
   value,
   onChange,
+  numOfPosts,
 }: Props) {
```
Comment (sk-kitsune) [inline containers/PageProjectDetails/containers/ProjectDetails/components/TabControl/index.tsx:22]: About meaning, you are changing `TabControl` (a dumb component)  to `TabControlInProjectDetails` (a 
smart component).

What are your reasons?

Comment (sk-kitsune) [reply containers/PageProjectDetails/containers/ProjectDetails/components/TabControl/index.tsx:22]: 🔴 **Blocking issue**: this PR cannot be merged until this issue is resolved.

**Review by sk-kitsune — CHANGES_REQUESTED** [review]
⛔ **Changes requested**: there are blocking issues.

**Comment by sk-tori** [comment]
@sk-kitsune @sk-tenba This is nice to have, yes, but the Asana task specifically mentions "in platform stats" (on the homepage). This is a completely different thing, solving a "Medium/High" task over a "Highest" task based on how user-facing each is.

**Comment by sk-tenba** [comment]
Since the amount of code to refactor is quite large, I will resolve this in the upcoming PR

**Pull request closed without merging**